Simplenote/Remote.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+class Remote {
+    enum Result: Equatable {
+        static func == (lhs: Remote.Result, rhs: Remote.Result) -> Bool {
+            switch (lhs, rhs) {
+            case (.success, .success):
+                return true
+            case (.failure(let code1, _), .failure(let code2, _)):
+                return code1 == code2
+            default:
+                return false
+            }
+
+        }
+
+        case success
+        case failure(_ statusCode: Int, _ error: Error?)
+    }
+
+    private let urlSession: URLSession
+
+    init(urlSession: URLSession = URLSession.shared) {
+        self.urlSession = urlSession
+    }
+
+    /// Send  task for remote
+    /// Sublcassing Notes: To be able to send a task it is required to first setup the URL request for the task to use
+    ///
+    func task(with request: URLRequest, completion: @escaping (_ result: Result) -> Void) {
```

Comment:
This API signature would imply that a `task` is returned.

WDYT about something like... `func performDataTask(with request: URLRequest, completion:..)` ?